// protobuf code generated by AST_protobuf.tpl
syntax = "proto3";



package swdv.test_types_service;

// Generic result message
message operation_result {
  bool result = 1;
}

// IFEX Namespace types
// IFEX Enum color_t
enum color_t {
    RED = 0;
    GREEN = 1;
    BLUE = 2;
    YELLOW = 3;
}

// IFEX Enum status_t
enum status_t {
    STATUS_OK = 0;
    STATUS_ERROR = 1;
    STATUS_PENDING = 2;
    STATUS_CANCELLED = 3;
}

// IFEX Struct point_2d_t
message point_2d_t {
  double x = 1;
  double y = 2;
}

// IFEX Struct point_3d_t
message point_3d_t {
  double x = 1;
  double y = 2;
  double z = 3;
}

// IFEX Struct sensor_data_t
message sensor_data_t {
  uint32 sensor_id = 1;
  uint64 timestamp = 2;
  float temperature = 3;
  float pressure = 4;
  float humidity = 5;
  point_3d_t location = 6;
  status_t status = 7;
  string error_message = 8;
}

// IFEX Struct test_all_primitives_t
message test_all_primitives_t {
  bool bool_val = 1;
  int32 int32_val = 2;
  int64 int64_val = 3;
  uint32 uint32_val = 4;
  uint64 uint64_val = 5;
  float float_val = 6;
  double double_val = 7;
  string string_val = 8;
  bytes bytes_val = 9;
}

// IFEX Struct nested_struct_t
message nested_struct_t {
  string id = 1;
  test_all_primitives_t primitives = 2;
  repeated point_3d_t points = 3;
  repeated color_t colors = 4;
  string metadata = 5;
}

// IFEX Method test_primitives
message test_primitives_request {
  test_all_primitives_t primitives = 1;
}

message test_primitives_response {
  test_all_primitives_t echoed = 1;
  bool validation_passed = 2;
}

service test_primitives_service {
  rpc test_primitives(test_primitives_request) returns (test_primitives_response);
}

// IFEX Method test_arrays
message test_arrays_request {
  repeated int32 int_array = 1;
  repeated string string_array = 2;
  repeated point_2d_t point_array = 3;
}

message test_arrays_response {
  int64 int_sum = 1;
  string concatenated_strings = 2;
  point_2d_t centroid = 3;
}

service test_arrays_service {
  rpc test_arrays(test_arrays_request) returns (test_arrays_response);
}

// IFEX Method test_enums
message test_enums_request {
  color_t color = 1;
  status_t status = 2;
}

message test_enums_response {
  string color_name = 1;
  string status_message = 2;
  color_t next_color = 3;
}

service test_enums_service {
  rpc test_enums(test_enums_request) returns (test_enums_response);
}

// IFEX Method test_nested_structs
message test_nested_structs_request {
  nested_struct_t nested = 1;
}

message test_nested_structs_response {
  string summary = 1;
  uint32 point_count = 2;
  uint32 color_count = 3;
}

service test_nested_structs_service {
  rpc test_nested_structs(test_nested_structs_request) returns (test_nested_structs_response);
}

// IFEX Method test_optional_fields
message test_optional_fields_request {
  string required_string = 1;
  int32 optional_int = 2;
  point_2d_t optional_struct = 3;
  repeated string optional_array = 4;
}

message test_optional_fields_response {
  uint32 received_count = 1;
  string summary = 2;
}

service test_optional_fields_service {
  rpc test_optional_fields(test_optional_fields_request) returns (test_optional_fields_response);
}

// IFEX Method test_sensor_data_stream
message test_sensor_data_stream_request {
  repeated sensor_data_t sensor_readings = 1;
  status_t filter_by_status = 2;
}

message test_sensor_data_stream_response {
  uint32 processed_count = 1;
  float average_temperature = 2;
  repeated uint32 error_sensors = 3;
  string summary_report = 4;
}

service test_sensor_data_stream_service {
  rpc test_sensor_data_stream(test_sensor_data_stream_request) returns (test_sensor_data_stream_response);
}





